app/src/main/java/nz/org/cacophony/birdmonitor/Util.java

```diff
@@ -784,58 +802,61 @@ public static void createCreateAlarms(Context context) { // Because each alarm n
         }
 
         alarmManager.setInexactRepeating(AlarmManager.ELAPSED_REALTIME_WAKEUP, AlarmManager.INTERVAL_FIFTEEN_MINUTES, AlarmManager.INTERVAL_DAY, pendingIntent);
+    }
 
+    public static long getMilliSecondsBetweenRecordings(Prefs prefs) {
+        if (prefs.getUseVeryFrequentRecordings()) {
+            return (long)prefs.getTimeBetweenVeryFrequentRecordingsSeconds();
+        }
+
+        float chance = new Random().nextFloat();
+        float shortWindowChance = prefs.getshortRecordingWindowChance();
+        float percent;
```

Comment:
Would be functional the same but you could have `float percentage = new Random().nextFloat();` instead of calculating the percentage from `chance`. Might make the code a bit clearer.

Reply:
Just realised don't even need to get a new random just use chance

Reply:
You can't use `chance` as it will always be from 0-0.25 for short windows and 0.25-1 for long windows.

Reply:
oh ye, silly me